File: minion/app/src/main/java/snakesoft/minion/WpModel.java
package snakesoft.minion;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;

/**
 * Created by snake on 6/10/18.
 */

public class WpModel
{
    public void loadState(LocalDatabase localDb) {
        mLocalDb = localDb;
        try
        {
            ByteArrayInputStream byteStream = new ByteArrayInputStream(localDb.get("WillPowerTracking"));
            mSuccessAtempts = byteStream.read();
            mFailedAtempts = byteStream.read();
        }
        catch (Exception ex)
        {
            mSuccessAtempts = 0;
            mFailedAtempts = 0;
        }
    }

    public void saveState()
    {
        ByteArrayOutputStream bos = new ByteArrayOutputStream();

        bos.write(mSuccessAtempts);
        bos.write(mFailedAtempts);

        mLocalDb.put("WillPowerTracking", bos.toByteArray());
    }

    public int getSuccessAtempts()
    {
        return mSuccessAtempts;
    }

    public int getFailedAtempts()
    {
        return mFailedAtempts;
    }

    public void setSuccessAtempts(int newValue)
    {
        mSuccessAtempts = newValue;
        saveState();
    }

    public void setFailedAtempts(int newValue)
    {
        mFailedAtempts = newValue;
        saveState();
    }

    private LocalDatabase mLocalDb;
    private int mSuccessAtempts = 0;
    private int mFailedAtempts = 0;
}

File: minion/app/src/main/java/snakesoft/minion/Models/WpModel.kt
package snakesoft.minion.Models

class WpModel (var Db: LocalDatabase)
{
    var SuccessAttempts: Int = 0
        set(newValue)
        {
            field = newValue
            saveState()
        }

    var FailedAttempts: Int = 0
        set(newValue)
        {
            field = newValue
            saveState()
        }

    init
    {
        loadState()
    }

    private fun loadState()
    {
        try
        {
            val splitted = Db["WillPowerTracking"].split(' ')
            SuccessAttempts = splitted[0].toInt()
            FailedAttempts = splitted[1].toInt()
        }
        catch (ex: Exception)
        {

        }

    }

    private fun saveState()
    {
        Db.put("WillPowerTracking", "$SuccessAttempts $FailedAttempts")
    }
}
